include/dxc/Support/dxcapi.use.h

```diff
@@ -31,15 +31,11 @@ class DxcDllSupport {
 
 #ifdef _WIN32
     m_dll = LoadLibraryA(dllName);
-#else
-    m_dll = ::dlopen(dllName, RTLD_LAZY);
-#endif
-
     if (m_dll == nullptr) return HRESULT_FROM_WIN32(GetLastError());
-
-#ifdef _WIN32
     m_createFn = (DxcCreateInstanceProc)GetProcAddress(m_dll, fnName);
 #else
+    m_dll = ::dlopen(dllName, RTLD_LAZY);
+    if (m_dll == nullptr) return E_FAIL;
     m_createFn = (DxcCreateInstanceProc)::dlsym(m_dll, fnName);
```

Comment:
I think we should do similar for dlsym() if you feel like it, you can use the error codes I suggested here: https://github.com/microsoft/DirectXShaderCompiler/pull/4713#issuecomment-1275447597